portality/store.py

```diff
@@ -175,8 +176,12 @@ def store(self, container_id, target_name, source_path=None, source_stream=None)
         if source_path:
             shutil.copyfile(source_path, tpath)
         elif source_stream:
-            with open(tpath, "w") as f:
-                f.write(source_stream.read())
+            data = source_stream.read(self.buffer_size)
+            mode = "w" if isinstance(data, str) else "wb"
```

Comment:
Could you please explain to me briefly why these data may be of different type?

Reply:
unfortunately in python 3 `read` may return `bytes` or `str`, which are treated very differently.  It depends on how the source stream was initialised - for example StringIO returns a string, but BytesIO returns bytes, meanwhile opening a file with "b" or without will change the return type.  Also, we take streams off HTTP, and I don't think we can control whether that is a string or a byte output.